extensions/search-router/src/search.tsx

```diff
@@ -21,6 +21,18 @@ export default async function search(props: LaunchProps<{ arguments: { query: st
       return;
     }
 
+    snap: if (query.includes("@")) {
+      const match = query.match(/@(\S+)/i);
+      const snapCandidate = match?.[1]?.toLowerCase();
+      if (!snapCandidate) break snap;
+
+      const searchEngine = searchEngines.find((engine) => engine.t === snapCandidate);
+      if (!searchEngine) break snap;
+
+      query = query.replace(/@\S+\s*/i, "").trim();
+      query += ` site:${searchEngine.ad || searchEngine.d}`;
+    }
+
     const cleanQuery = query.replace(/!\S+\s*/i, "").trim();
```

Comment:
I think we could instead always replace both @-parts and !-parts directly here, and keep `const query` in the top. Then we can just in one const check if we have a @-part. Better to extract this to a function or similar to avoid the unusual label flow